torch/csrc/Exceptions.h

```diff
@@ -77,6 +79,11 @@ static inline void PyErr_SetString(PyObject* type, const std::string& message) {
       PyErr_SetString(PyExc_NotImplementedError, torch::processErrorMsg(msg)); \
       retstmnt;                                                      \
     }                                                                \
+    catch (const c10d::TimeoutException& e) {                        \
```

Comment:
curious about how did we determine the order of the catch clauses here?

Reply:
I believe we can move `TimoutException` below `c10::Error`, since all other clauses above it are subclasses of `c10::Error`. We can have `TimeoutException` and `system_error` at the end of the list.